examples/vision/attention_mil_classification.py

```diff
@@ -0,0 +1,600 @@
+"""
+Title: Classification using Attention-based Deep Multiple Instance Learning (MIL).
+Author: [Mohamad Jaber](https://www.linkedin.com/in/mohamadjaber1/)
+Date created: 2021/08/16
+Last modified: 2021/11/13
+Description: MIL approach to classify bags of instances and get their individual instance score.
+"""
+"""
+## Introduction
+
+### What is Multiple Instance Learning (MIL)?
+Usually for supervised learning  algorithms, the learner receives labels for a set of
+instances. In the case of MIL, the learner receives labels for a set of bags in which each
+bags contains a set of instances. The bag is labelled positive if it contains at least
+one positive instance and negative if it does not contain any.
+
+#### Motivation
+It is often assumed in image classification tasks that each image clearly represents a
+class label. In medical imaging (e.g. computational pathology, etc.) an *entire image*
+is represented by a single class label (cancerous/non-cancerous) or a region of interest
+could be given. However, one will be interested in knowing which patterns in the image
+is actually causing it to belong to that class. In this context, the image(s) will be
+divided and the subimages will form the bag of instances.
+
+Therefore, the goals are to:
+
+1. learn a model to predict class label of a bag of instances.
+2. know the score of the instances within the bag which resulted to the class label
+prediction.
+
+#### Implementation
+
+The end-to-end model is composed of:
+
+1. The backbone (feature extractor layers) of the model.
+2. The extracted features fed into the MIL attention layer. The layer is modelled
+as permutation-invariant.
+3. The attention scores of each feature multipled by their respective input features
+(instances).
+4. The multipled features are passed to a softmax function for getting the vector of
```

Comment:
These 4 points are confusing: "features" or "attention scores" are not model components. They're outputs. Model components are layers / blocks of layers.

Reply:
I rephrased it in my recent [commit](https://github.com/keras-team/keras-io/pull/593/commits/b2071513d2d32b399564883ff81cfa8dee5bf84d).
I wanted to emphasize on the process of model's implementation rather than merely describing the components.